src/common/googleCloud.ts

```diff
@@ -29,3 +32,129 @@ export const downloadJsonFile = async <T>({
 
   return JSON.parse(result);
 };
+
+export enum UserActiveState {
+  Active = '1',
+  InactiveSince6wAgo = '2',
+  InactiveSince12wAgo = '3',
+  NeverActive = '4',
+}
+
+export const userActiveStateQuery = `
+  with d as (
+    select u.primary_user_id,
+    min(last_app_timestamp) as last_app_timestamp,
+    min(registration_timestamp) as registration_timestamp,
+    min(
+        case
+          when period_end is null then '4'
+          when period_end between date(@previous_date - interval 6*7 day) and @previous_date then '1'
+          when period_end between date(@previous_date - interval 12*7 day) and date(@previous_date - interval 6*7 + 1 day) then '2'
+          when date(u.last_app_timestamp) <  date(@previous_date - interval 12*7 day) then '3'
+          when date(u.registration_timestamp) <  date(@previous_date - interval 12*7 day) then '3'
+          else '4' end
+      ) as previous_state,
+      min(
+        case
+            when period_end is null then '4'
+            when period_end between date(@run_date - interval 6*7 day) and @run_date then '1'
+            when period_end between date(@run_date - interval 12*7 day) and date(@run_date - interval 6*7 + 1 day) then '2'
+            when date(u.last_app_timestamp) <  date(@run_date - interval 12*7 day) then '3'
+            when date(u.registration_timestamp) <  date(@run_date - interval 12*7 day) then '3'
+            else '4' end
+      ) as current_state,
+      min(
+        case
+          when period_end is null then '4. never_active'
+          when period_end between date(@previous_date - interval 6*7 day) and @previous_date then '1. active_last_6w'
+          when period_end between date(@previous_date - interval 12*7 day) and date(@previous_date - interval 6*7 + 1 day) then '2. active_7w_12w'
+          when date(u.last_app_timestamp) <  date(@previous_date - interval 12*7 day) then '3. active_12w+'
+          when date(u.registration_timestamp) <  date(@previous_date - interval 12*7 day) then '3. active_12w+'
+          else '4. never_active' end
```

Comment:
I think Dave forgot to change the values here to numbers only

Reply:
We have that, way above, he duplicated the lines to have both.

Reply:
Are you sure? why is it needed? Please double check

Reply:
Yes, we can just remove these extra columns. Maybe I will just then add a comment up above to define the values. Or the enum itself is the definition.